### PR TITLE
PICA-136: upgrade to v0.9.33

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**
#### Intro

Upgrade from v0.9.30 to v0.9.33

#### Migrations

- v0.9.31
  - [x] https://github.com/paritytech/polkadot/pull/6091
  - [x] https://github.com/paritytech/polkadot/pull/5729
    - [x] Preimage v1
    - [x] Scheduler v4
    - [x] Democracy v1
    - [x] Multisig v1
- v0.9.32
- v0.9.33

#### Notes

Are we forced to have the old migrations listed? Parity removed the v2 to v3 migration for the Scheduler, I changed it to a combination of v1 to v4 and v3 to v4 as they are idempotent.

I removed `price-feed` and associated packages as they are outdated and not particularly useful. 

The `local-integration-tests` package must be updated.

### CODE REVIEW

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -1028,27 +1028,28 @@ jobs:
...
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
-          persist-credentials: false
-      - uses: "./.github/templates/watch-exec"
-        with:
-          command: nix -- build .#price-feed
-          token: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+  # FIXME(hussein): enable when price feed is upgraded and actually usefull
```
Comment (vimukthi-git) [inline .github/workflows/build-test-deploy.yml:1031]: :+1: 

We can tie this to Apollo initiative when we have one

File: code/parachain/runtime/common/src/lib.rs [hunk 2]
```diff
@@ -135,7 +135,9 @@ mod constants {
 	pub const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(75);
 
 	/// We allow for 2 seconds of compute with a 6 second average block time.
-	pub const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND.saturating_div(2);
+	pub const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND
+		.saturating_div(2)
+		.set_proof_size(polkadot_primitives::v2::MAX_POV_SIZE as u64);
```
Comment (vimukthi-git) [inline code/parachain/runtime/common/src/lib.rs:140]: Wonder what this does

**Review by vimukthi-git — COMMENTED** [review]
LGTM, I will check a local upgrade on a fork for this.

**Comment by vimukthi-git** [comment]
>Are we forced to have the old migrations listed?

Unfortunately for Pablo we need them to be executed to have the pools in the runtime when testing for now. We will change that in the release 4 after removing the assets hard coding.

File: code/parachain/runtime/common/src/lib.rs (hunk 2 above)
Comment (hussein-aitlahcen) [reply code/parachain/runtime/common/src/lib.rs:140]: Weight V2 introduce a new `proof_size` field in weights, this just define it's max size as being the max proof of validity size we submit to the relay

File: code/Cargo.toml [hunk 3]
```diff
@@ -42,9 +42,10 @@ members = [
   "parachain/runtime/composable",
   "parachain/runtime/composable-wasm",
   "utils/common",
-  "utils/price-feed",
+  # FIXME(hussein): this package is outdated and broken
+  # "utils/price-feed",
   "utils/collator-sidecar",
-  "utils/composable-subxt",
+  # "utils/composable-subxt",
   # TODO(hussein): do we completely remove this?
```
Comment (KaiserKarel) [inline code/Cargo.toml:49]: yeah let's completely remove

**Review by obsessed-cake — APPROVED** [review]

**Comment by vimukthi-git** [comment]
Upgrade on a fork of Picasso works fine. Did not check XCM though. We can check it after making a RC.

**Review by vimukthi-git — APPROVED** [review]

**Comment by KaiserKarel** [comment]
Task linked: [PICA-136 Update main code base](https://app.clickup.com/t/20465559/PICA-136)

**Review by obsessed-cake — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
> The local-integration-tests package must be updated.

https://github.com/ComposableFi/composable/pull/3015

File: code/parachain/runtime/composable/src/lib.rs [hunk 4]
```diff
@@ -855,7 +861,12 @@ pub type Executive = executive::Executive<
 	system::ChainContext<Runtime>,
 	Runtime,
 	AllPalletsWithSystem,
-	SchedulerMigrationV3,
+	(
+		SchedulerMigrationV1toV4,
+		preimage::migration::v1::Migration<Runtime>,
+		scheduler::migration::v3::MigrateToV4<Runtime>,
```
Comment (vivekvpandya) [inline code/parachain/runtime/composable/src/lib.rs:867]:
```rust
  3     impl<T: Config<Hash = PreimageHash>> OnRuntimeUpgrade for MigrateToV4<T> {                                                         
  2         #[cfg(feature = "try-runtime")]                                                                                                
  1         fn pre_upgrade() -> Result<Vec<u8>, &'static str> {                                                                            
101             assert_eq!(StorageVersion::get::<Pallet<T>>(), 3, "Can only upgrade from version 3");
```
Just curious above will hit assert has migrate_v1_to_v4() sets version to 4, unless executed without assert or may be directly execute migrate_v1_to_v4()

**Pull request merged**